pyccel/stdlib/numpy/numpy_f90.f90

```diff
@@ -0,0 +1,156 @@
+    !........................................
+  
+    !........................................
+    elemental function numpy_sign_i32(x) result(Out_0001)
+  
+      implicit none
+  
+      integer(i32) :: Out_0001
+      integer(i32), value :: x
+  
+      Out_0001 = merge(0_i32, (merge(1_i32, -1_i32, x > 0_i32)), x == 0_i32)
+      return
+  
+    end function numpy_sign_i32
+    !........................................
+  
+    !........................................
+    elemental function numpy_sign_i64(x) result(Out_0001)
+  
+      implicit none
+  
+      integer(i64) :: Out_0001
+      integer(i64), value :: x
+  
+      Out_0001 = merge(0_i64, (merge(1_i64, -1_i64, x > 0_i64)), x == 0_i64)
+      return
+  
+    end function numpy_sign_i64
+    !........................................
+  
+    !........................................
+    elemental function numpy_sign_f32(x) result(Out_0001)
+  
+      implicit none
+  
+      real(f32) :: Out_0001
+      real(f32), value :: x
+  
+      Out_0001 = merge(0_f32, (merge(1_f32, -1_f32, x > 0_f32)), x == 0_f32)
+      return
+  
+    end function numpy_sign_f32
+    !........................................
+  
+    !........................................
+    elemental function numpy_sign_f64(x) result(Out_0001)
+  
+      implicit none
+  
+      real(f64) :: Out_0001
+      real(f64), value :: x
+  
+      Out_0001 = merge(0_f64, (merge(1_f64, -1_f64, x > 0_f64)), x == 0_f64)
+      return
+  
+    end function numpy_sign_f64
+    !........................................
+  
+    !........................................
+    elemental function numpy_sign_c32(x) result(Out_0001)
+  
+      implicit none
+
+      complex(c32) :: Out_0001
+      complex(c32), value :: x
+      logical :: x_ne_zero ! Condition for x different than 0
+      logical :: x_lt_zero ! Condition for x less than 0
+  
+      x_ne_zero = (REALPART(x) .ne. 0_f32) .or. (IMAGPART(x) .ne. 0_f32)
+      x_lt_zero = ((REALPART(x) .eq. 0_f32) .and. IMAGPART(x) .lt. 0_f32) &
+              .or. (REALPART(x) .lt. 0_f32)
+  
+      Out_0001 = merge(merge(-1_c32, 1_c32, x_lt_zero), 0_c32, x_ne_zero)
+      return
+  
+    end function numpy_sign_c32
+    !........................................
+  
+    !........................................
+    elemental function numpy_sign_c64(x) result(Out_0001)
+  
+      implicit none
+  
+      complex(c64) :: Out_0001
+      complex(c64), value :: x
+      logical :: x_ne_zero ! Condition for x different than 0
+      logical :: x_lt_zero ! Condition for x less than 0
+
+      x_ne_zero = (REALPART(x) .ne. 0_f64) .or. (IMAGPART(x) .ne. 0_f64)
+      x_lt_zero = ((REALPART(x) .eq. 0_f64) .and. IMAGPART(x) .lt. 0_f64) &
+              .or. (REALPART(x) .lt. 0_f64)
+
+      Out_0001 = merge(merge(-1_c64, 1_c64, x_lt_zero), 0_c64, x_ne_zero)
+      return
+
+    end function numpy_sign_c64
+    !........................................
+  
+  end module numpy_f90
```

Comment:
No newline at end of file

Reply:
https://github.com/pyccel/pyccel/pull/1169/commits/2a644c9e99378e351e50faea29acd54ef1e34f9a